### PR TITLE
Better pinned ui

### PR DESCRIPTION
**Pull request by g-harel**
screenshot_1519873379

screenshot_1519873389

### CODE REVIEW

**Review by erinb90 — APPROVED** [review]
Looks really nice! 😀

File: src/org/thoughtcrime/securesms/PinnedMessageAdapter.java [hunk 1]
```diff
@@ -107,22 +109,36 @@ public void onBindViewHolder(ViewHolder holder, int position) {
...
+                dialog.cancel();
+            });
+
+            builder.setNegativeButton(R.string.no, (dialog, id) -> dialog.cancel());
+            builder.create().show();
+            return true;
         });
     }
 
     private void setMessageView(MessageRecord record, ViewHolder viewHolder) {
```
Comment (shelbaz) [inline src/org/thoughtcrime/securesms/PinnedMessageAdapter.java:131]: Awesome, much clearer now if message is from sender or receiver .

**Comment by shelbaz** [comment]
Looks great. Does ur expresso test include testing both Pin and unPin on both conversation screen and pinned messages screen? I cant tell from looking at [PinnedMessageEspressoTest.java](https://github.com/Radu-Raicea/SignalAndroid/blob/master/test/androidTest/java/org/thoughtcrime/securesms/PinnedMessageEspressoTest.java)

**Comment by g-harel** [comment]
@shelbaz 

The pin/unpin tests already existed in the conversation screen.

It is not possible to pin from the "pinned messages" screen, so that is not tested. 

I did however update the test for the unpin function on that screen.

https://github.com/Radu-Raicea/SignalAndroid/blob/267838fb55ad620ba754da9a6075a482cf596899/test/androidTest/java/org/thoughtcrime/securesms/PinnedMessageEspressoTest.java#L115

**Review by shelbaz — APPROVED** [review]
Got it, i understand the tests now.

**Pull request merged**